app/javascript/src/PaymentGateways/braintree/BraintreeForm.jsx

```diff
@@ -0,0 +1,148 @@
+/* eslint-disable no-console */
+// @flow
+
+import React, { useState, useEffect, useRef } from 'react'
+import {
+  BraintreeUserFields,
+  BraintreeCardFields,
+  BraintreeBillingAddressFields,
+  BraintreeSubmitFields,
+  createHostedFieldsInstance,
+  create3DSecureInstance,
+  validationConstraints,
+  hostedFieldOptions,
+  veryfyCard
+} from 'PaymentGateways'
+import hostedFields from 'braintree-web/hosted-fields'
+import threeDSecure from 'braintree-web/three-d-secure'
+import { CSRFToken } from 'utilities/utils'
+import validate from 'validate.js'
+import './styles.css'
+
+import type { Node } from 'react'
+import type { BraintreeFormProps } from 'PaymentGateways'
+
+const BraintreeForm = ({
+  braintreeClient,
+  billingAddress,
+  threeDSecureEnabled,
+  formActionPath,
+  countriesList,
+  selectedCountryCode
+}: BraintreeFormProps): Node => {
+  const formRef = useRef<HTMLFormElement | null>(null)
+  const [hostedFieldsInstance, setHostedFieldsInstance] = useState(null)
+  const [braintreeNonceValue, setBraintreeNonceValue] = useState('test-mode-placeholder')
+  const [billingAddressData, setBillingAddressData] = useState(billingAddress)
+  const [isCardValid, setIsCardValid] = useState(false)
+  const [isFormValid, setIsFormValid] = useState(false)
+  const [formErrors, setFormErrors] = useState('Form is empty')
+  const [cardError, setCardError] = useState(null)
+  const [isAwaiting, setIsAwaiting] = useState(false)
+
+  useEffect(() => {
+    const getHostedFieldsInstance = async () => {
+      const HFInstance = await createHostedFieldsInstance(hostedFields, braintreeClient, hostedFieldOptions, setIsCardValid)
+      setHostedFieldsInstance(HFInstance)
+    }
+    getHostedFieldsInstance()
+  }, [])
+
+  useEffect(() => {
+    if (braintreeNonceValue !== 'test-mode-placeholder' && formRef.current) {
```

Comment:
Do we have another way to do this without `test-mode-placeholder` ?
Why is initial value like a String? Can't it be `null`?

```js
  const [braintreeNonceValue, setBraintreeNonceValue] = useState(null)

```